extensions/powershell/powershell.discover.ps1

```diff
@@ -0,0 +1,25 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+[CmdletBinding()]
+param ()
+
+begin {
+    $psPaths = $env:PSModulePath -split [System.IO.Path]::PathSeparator | Where-Object { $_ -notmatch 'WindowsPowerShell' }
+}
+process {
+    $manifests = $psPaths | ForEach-Object -Parallel {
+        $searchPatterns = @('*.dsc.resource.json', '*.dsc.resource.yaml', '*.dsc.resource.yml')
+        $enumOptions = [System.IO.EnumerationOptions]@{ IgnoreInaccessible = $false; RecurseSubdirectories = $true }
```

Comment:
I'm concerned about the perf impact.  Can we have some caching that stores the last modified date of a PSModulePath segment and only search it if it has changed from last time?  This also requires currently discovered ones to be persisted on disk.